Phylanx Release build fails to link

When building Phylanx with gcc 7.3 (Release build), pybind11 is adding additional flags that break linking.  This is the error:  (see http://omega.nic.uoregon.edu:8020/#/builders/2/builds/57 for example)

```
[ 60%] Linking CXX shared module ../../bin/_phylanx.cpython-36m-x86_64-linux-gnu.so
/bin/ld: /tmp/ccvkdP3c.ltrans12.ltrans.o: relocation R_X86_64_PC32 against undefined symbol `_ZNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEED1Ev' can not be used when making a shared object; recompile with -fPIC
/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
make[2]: *** [bin/_phylanx.cpython-36m-x86_64-linux-gnu.so] Error 1
make[1]: *** [python/src/CMakeFiles/phylanx_py.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

A workaround is to use the `NO_EXTRAS` setting for the `pybind11_add_module()` call in the `python/src/CMakeLists.txt` file.  Not sure if this is a "fix" though.

--- a/python/src/CMakeLists.txt
+++ b/python/src/CMakeLists.txt
@@ -59,7 +59,7 @@ if(APPLE) # OSX requires this to be a shared library, not a module (default)
       ${phylanx_py_SOURCES}
       ${phylanx_py_HEADERS})
 else(APPLE)
-  pybind11_add_module(phylanx_py
+  pybind11_add_module(phylanx_py NO_EXTRAS
     ${phylanx_py_SOURCES}
     ${phylanx_py_HEADERS}
     ${phylanx_python_SOURCES})

